lib/bacnet-asn1.js

```diff
@@ -779,7 +784,14 @@ var decode_read_access_result = module.exports.decode_read_access_result = funct
         var localResult = bacapp_decode_application_data(buffer, offset + len, apduLen + offset - 1, value.objectIdentifier.type, new_entry.propertyIdentifier);
         if (!localResult) return;
         len += localResult.len;
-        localValues.push({value: localResult.value, type: localResult.type});
+        var resObj = {
+          value: localResult.value,
+          type: localResult.type
+        };
+        if (localResult.encoding !== undefined) {
```

Comment:
I'm not entirely sure, if it's a smart idea to have checks in the generic encoding function, targeting encodings explicitly. It might be a better idea to wrap string in a value object, similar to the bit-string object. E.g:

``` js
var stringResult = {value: 'Test1234$', encoding: 1};
```